packages/yourloops/pages/patient-view/target-and-alerts/target-and-alerts-view.tsx

```diff
@@ -0,0 +1,110 @@
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { FC, useEffect, useState } from 'react'
+import Container from '@mui/material/Container'
+import MonitoringAlertsContentConfiguration
+  from '../../../components/monitoring-alert/monitoring-alerts-content-configuration'
+import { Patient } from '../../../lib/patient/models/patient.model'
+import { MonitoringAlertsParameters } from '../../../lib/team/models/monitoring-alerts-parameters.model'
+import { usePatientsContext } from '../../../lib/patient/patients.provider'
+import { useAlert } from '../../../components/utils/snackbar'
+import { useTranslation } from 'react-i18next'
+import { useSelectedTeamContext } from '../../../lib/selected-team/selected-team.provider'
+import Card from '@mui/material/Card'
+import { useTheme } from '@mui/material/styles'
+import CardHeader from '@mui/material/CardHeader'
+import CardContent from '@mui/material/CardContent'
+import Typography from '@mui/material/Typography'
+import { useLocation } from 'react-router-dom'
+
+interface TargetAndAlertsViewProps {
+  patient: Patient
+}
+
+export const MONITORING_ALERTS_SECTION_ID = 'monitoring-alerts'
+
+export const TargetAndAlertsView: FC<TargetAndAlertsViewProps> = (props) => {
+  const { patient } = props
+  const theme = useTheme()
+  const { t } = useTranslation('yourloops')
+  const { updatePatientMonitoringAlertsParameters } = usePatientsContext()
+  const { selectedTeam } = useSelectedTeamContext()
+  const alert = useAlert()
+  const { pathname, hash, key } = useLocation()
+  const monitoringAlertsParameters = patient.monitoringAlertsParameters ?? selectedTeam.monitoringAlertsParameters
+
+  const [saveInProgress, setSaveInProgress] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (hash === '') {
+      return
+    }
+
+    const sectionId = hash.replace('#', '')
+    const section = document.getElementById(sectionId)
```

Comment:
It is not advised in react to access/manipulate the dom using `document` (preferred way is to use refs).
More info: [here](https://www.shecodes.io/athena/43520-how-to-use-getelementbyid-with-react#:~:text=getElementById%20cannot%20be%20used%20directly,with%20the%20element%20that%20way.) and [here](https://stackoverflow.com/questions/72591035/can-i-use-the-document-object-in-react#:~:text=You%20can%2C%20but%20this%20goes,that%20is%20cross%2Dbrowser%20compatible.)

I don't think that it is much of an issue here (if I understood everything correctly), so I think that it is ok to keep as is if you don't feel like using refs.

Reply:
I replaced it and used a ref